### ISSUE TITLE
Libstreaming Status

### ISSUE THREAD

**Issue by deshan** [fyhertz/libstreaming#251]
Hi All,

Does this project stopped? Looks No activities for years. If deprecated any good alternative?

Thanks
Deshan

**Comment by KPAXIT** [comment]
Hi,

I also would like to know if there are some plans and/or intentions to continue with this project, either by original maker 'fyhertz' or community?

From what I know is that 
- there is no activity for 2 years (last commit  was Oct 2015)
- up to Lollipop it's still usable (maybe with some workarounds) but android 6.0+ has a lot of problems, and stackoverflow is full of related questions without any proper solution, at least I couldn't find one...

Now I'm also looking for alternative, maybe we can use this ticket/topic to share some and discuss? 
I found recently this one https://github.com/begeekmyfriend/yasea and checking it out. 
Some small findings about this one & example app: 
-uses RTMP protocol
-works on 6.0+
-max smooth resolution I could get was 1280x720 ~24fps hardware-encoded, 1080 not working well (not sure what is a limiting factor here, devices should be able to handle it )
-no adaptive bitrate (jet?)
-camera preview works only while streaming/recording in example app
-some issues: onOrientationChange, switching camrea and/or while publishing, sometimes I need to republish to make streaming work again, otherwise its black 

PS I really was exited about libstreaming but current status doesn't give any confidence to use it in project(s) for production, so I'm looking hard for some options/solutions (eg RTSP, webRTC, gstreamer, ffmpeg, etc)  to make livestreaming with reasonable delay(in my case) work. Any input on this one would be very appreciated.

**Comment by fyhertz** [comment]
Hi folks,

Indeed ! I am not working on libstreaming anymore. I was a student when I started the project as a side activity. I am not dead, just working now... For a while I thought I would get back working on it at some point, but let's face it, it's not going to happen. You can consider everything here is public domain now. 

Anyway, I would love to see people saving the project !

I should really update the Readme to make that clear...

Regards,

**Comment by deshan** [comment]
Indeed sad to hear that such a good project is discontinued. Anyway it's up to the owner. But still it would be useful if the license is LGPL instead GPL.

Thanks for the great work shared with the community.
Deshan

**Comment by T0in0u** [comment]
@fyhertz Very sad that this project is dead, but get it (student life is easier to work on sides like this one). But I'm wondering something: could you turn the license under LGPL like @deshan said, to make this lib available for many projects and maybe, find some people reasoned to work on? I mean, GPL is too strict and commercial license is no longer alive anymore (old and abandoned project). Is this possible?

**Comment by spyhunter99** [comment]
How about the asf 2.0?

**Comment by T0in0u** [comment]
@spyhunter99 Yes, Apache License 2.0 will be great too!

**Comment by fyhertz** [comment]
I like the idea.

Which would you prefer? Apache or LGPL? I don't know much about the differences between the two.

**Comment by spyhunter99** [comment]
Apache is the most permissive. GPL flavored versions require anything that
depends on (either dynamically or statically linked) this library to also
be open source.

**Comment by fyhertz** [comment]
I agree that the GPL is too restrictive, my question was about choosing between the LGPL and the Apache 2.0... Or something else that allows close source usage.

**Comment by T0in0u** [comment]
I think LGPL and Apache 2.0 are appropriate.

Le 13 mars 2017 6:27 PM, Simon <notifications@github.com> a écrit :

I agree that the GPL is too restrictive, my question was about choosing between the LGPL and the Apache 2.0... Or something else that allows close source usage.

**Comment by sbaar** [comment]
I would recommend apache.

**Comment by sbaar** [comment]
Apache would almost guarantee the survival of the library. LGPL would probably kill it for good.

**Comment by deshan** [comment]
@fyhertz thanks for your decision for change the license less restrictive. +1 for apache

**Comment by JayasenaMS** [comment]
I would also recommend apache.

**Comment by hs79hs** [comment]
Thanks +1

**Comment by Tariq2518** [comment]
Libstreaming have a lot of issues. I will recommend you to use Gstreamer it is good if you want to do the same thing libstreaming is doing.
